### ISSUE TITLE
Consider changing For Each block

### ISSUE THREAD

**Issue by ZenithRogue** [LLK/scratch-blocks#1802]
While it has long been considered an obsolete block, the For Each block is still included in 3.0, and would still be very useful. I propose that we change the visual syntax of the block, as to not confuse scratchers are unfamiliar with it (should it be used in a project)

**Before:**
```
for each [variable] in (number) {
   say (variable)
}
```
**After:**
```
repeat (number) and advance [variable] {
   say (variable)
}
```

**Comment by towerofnix** [comment]
Here is kchadha's official word on changes to the "for each" block: https://github.com/LLK/scratch-vm/pull/963#issuecomment-375442701

>Since this block is being added to Scratch 3.0 specifically for maintaining compatibility with older projects, I think the behavior of the 3.0 block should have a one-to-one correspondence with the Scratch 2.0 block. Thus, I don't think we should change any behavior from the 2.0 block or add any features that the 2.0 block does not have.

But for the sake of discussion here, we'll assume that it's okay to change up the block.

---

>the For Each block [would] still be very useful.

What are your ideas of how it could still be useful? The current behavior of the for-each block is extremely easy to work around:

```
set [variable] to (1)
repeat (n)
    script
    change [variable] by (1)
end
```

And in my opinion this is still easier to understand than even "repeat (n) and advance (variable)".

**Comment by ZenithRogue** [comment]
The For block is quick and easy for shorthand coding, though I personally use the standard repeat function you show above due to accessibility.  I mostly think this change would make it easier for scratchers to understand what the block does, in the event that someone uses it in their code.

**Comment by TheLogFather** [comment]
I can't say I like the syntax in either case. The former is only there because that's more like other programming languages (which doesn't necessarily mean it's 'good', of course...)

I think a more natural way to say it might be something like: "count to (number) using [var]"
The fact it's a C-block should hopefully make it clear that it's a loop.

A variation that provides a little extra flexibility might be: "count from (start) to (end) using [var]"
(–that would also allow 'end' to be less than 'start', i.e. a countdown).

Of course, you could add a 'skip', too, etc., etc... (but I think that starts to get unwieldy)

**Comment by towerofnix** [comment]
>The fact it's a C-block should hopefully make it clear that it's a loop.

So would the "loop" indicator at the bottom of the C-block:

![The bottom of a generic loop C-block, showing the "loop" indicator, a curvy arrow pointing back towards the top of the loop C-block](https://user-images.githubusercontent.com/9948030/50013054-8537bd00-ff96-11e8-8187-86c5c63cd99d.png)

Aside: This is somewhat related to #1800 (if that were implemented, users could create their own "count from A to B" custom block).

**Comment by ZenithRogue** [comment]
@TheLogFather's definitely seems the simplest
![scratchblocks](https://user-images.githubusercontent.com/11393734/50014722-5582b800-ff8a-11e8-8e6c-158257d01242.png)

**Comment by cyx2015s** [comment]
I agree. I didn't know how to translate the block 'for each' into Chinese...
